javascript-typing-tutor/main.js

```diff
@@ -0,0 +1,36 @@
+
+var $span = document.querySelectorAll('span');
+var index = 0;
+var $incorrect = 0;
+var $hidden = document.querySelector('.hidden');
+var lettersTyped = 0;
+function handleKeyDown(event) {
+  lettersTyped++;
+  if ($span[index].textContent === event.key && index < 29) {
```

Comment:
Instead of hard-coding in the number `29`, you should use the length of the `$span` NodeList as your stop point. That way, if you ever change the phrase to be typed, your code will still work.